File: app/src/main/java/com/androidacademy/msk/exerciseproject/screen/intro/IntroView.java
package com.androidacademy.msk.exerciseproject.screen.intro;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndStrategy;
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

public interface IntroView extends MvpView {

    @StateStrategyType(AddToEndStrategy.class)
    void setLayout();

    @StateStrategyType(OneExecutionStateStrategy.class)
    void startNewsListActivity();
}

File: app/src/main/java/com/androidacademy/msk/exerciseproject/screen/intro/IntroView.kt
package com.androidacademy.msk.exerciseproject.screen.intro

import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndStrategy
import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

interface IntroView : MvpView {

    @StateStrategyType(AddToEndStrategy::class)
    fun setLayout()

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun startNewsListActivity()
}
